packages/devtools_app/lib/src/initializer.dart

```diff
@@ -113,16 +116,29 @@ class _InitializerState extends State<Initializer>
   /// Shows a "disconnected" overlay if the [service.serviceManager] is not currently connected.
   void _handleNoConnection() {
     WidgetsBinding.instance.addPostFrameCallback((_) {
-      if (!_checkLoaded() && ModalRoute.of(context).isCurrent) {
+      if (!_checkLoaded() &&
+          ModalRoute.of(context).isCurrent &&
+          currentDisconnectedOverlay == null) {
         Overlay.of(context).insert(_createDisconnectedOverlay());
+
+        // Set up a subscription to hide the overlay if we become reconnected.
+        disconnectedOverlayReconnectSubscription = serviceManager.onStateChange
```

Comment:
make sure to cancel the subscription in the dispose method or use the auto dispose mixin.